.github/workflows/deploy_on_push.yml

```diff
@@ -0,0 +1,40 @@
+run-name: ${{ github.actor }} deployed app
+on:
+  push:
+    branches:
+    - main
+    
+jobs:
+  test-flask:
```

Comment:
Separate the test and the deploy into separate workflows. That way you can just reuse the test one between your pr and push action l